frontend/components/commons/dataset-description/DatasetDescription.component.vue

```diff
@@ -1,30 +1,96 @@
 <template>
   <div class="description">
     <h2 class="--heading5 --semibold description__title">{{ title }}</h2>
-    <p class="--body1 description__text">{{ datasetDescription }}</p>
+    <div class="description__markdown" v-if="!isEditing">
+      <!-- Markdown content here  -->
+      <p class="--body1 description__text" v-html="description" />
+      <BaseButton name="edit-button" @on-click="onEdit">
+        <svgicon width="15" height="15" name="pen"></svgicon
+      ></BaseButton>
+    </div>
+    <div class="description__markdown__edit" v-if="isEditing">
+      <!-- Markdown input component here -->
+
+      <div class="description__button-area">
+        <base-button name="save-button" class="primary small" @on-click="onSave"
+          >Save</base-button
+        >
+        <base-button
+          name="close-button"
+          class="secondary small"
+          @on-click="onClose"
+          >Discard</base-button
+        >
+      </div>
+    </div>
   </div>
 </template>
 
 <script>
+import { getDatasetDescription, saveDescription } from "./MockedService";
+
 export default {
   props: {
-    datasetDescription: {
+    datasetId: {
+      type: Array,
+      required: true,
+    },
+    datasetTask: {
       type: String,
       required: true,
     },
   },
   data: () => {
     return {
       title: "Description and annotation guidelines",
+      isEditing: false,
     };
   },
+  computed: {
+    description() {
+      return getDatasetDescription(this.datasetId, this.datasetTask, false);
+    },
```

Comment:
1/ Instead of using the mock service, you can create the corresponding vuex orm query and call it here instead 
2/ performance : since the description will be update only in this component, it's not necesary to use a computed to fetch the description. You can get instead the data from the ORM in a hook lyfecycle instead (here an interesting [article](https://dev.to/pecus/when-to-avoid-vuejs-computed-properties-for-greater-performance-5anj))

Reply:
Thanks Keith;
About the first point, I'm not sure which part of dataset entity I must show, because i don't have so much experience in this project, sorry, if you want, we can pair programming and i'll be happy to understand where the data is.

Related to second point, I created computed prop because the user can update that value; other approach is get the value as a prop, but in my humble opinion, when the components has a dependency from external services like a rest api, is more easy to maintain when state is in the current component. 

Wait your response!
Thanks,